src/client.rs

```diff
@@ -206,26 +204,26 @@ where
     /// Update TUF metadata from the remote repository.
     ///
     /// Returns `true` if an update occurred and `false` otherwise.
-    pub fn update(&mut self) -> Result<bool> {
-        let r = self.update_root()?;
-        let ts = self.update_timestamp()?;
-        let sn = self.update_snapshot()?;
-        let ta = self.update_targets()?;
+    pub async fn update(&mut self) -> Result<bool> {
+        let r = await!(self.update_root())?;
```

Comment:
This will execute each of these updates in series-- is that what you want, or should the updates be able to run concurrently?

Reply:
Yeah, this serialization is mandated by the [tuf spec](https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#5-detailed-workflows). Each metadata depends on the last. The methods internally will skip actually downloading if it's not necessary.